pkg/ast/spl/spl.peg

```diff
@@ -4365,14 +4415,56 @@ Offset <- off:(RelIntegerAsString) tuOff:(RelTimeUnit) {
     return relTimeOffset, nil
 }
 
+ChainedRelativeTimestamp <- first:(Offset / Snap) rest:((Offset / Snap)*) {
+    var offsets []ast.RelativeTimeOffset
+    var snaps []string
+
+    // Append zero values to ensure corresponding offsets and snaps
+    // This ensures that each snap and offset correspond to the same index
+    appendZeroValues := func() {
+        if len(offsets) > len(snaps) {
+            snaps = append(snaps, "")
+        } else if len(snaps) > len(offsets) {
+            offsets = append(offsets, ast.RelativeTimeOffset{})
+        }
+    }
```

Comment:
It seems that you are assigning default values to the `offsets` and `snaps` slices. So I think it is better to rename it to `appendDefaultValues` or something like that, unless there is a specific reason why it is name as `appendZeroValues`.

Reply:
Addressed it in the latest commit.